### PR TITLE
Gcc warnings ptr to int

### PR DESCRIPTION
**Pull request by ShFil119**

### CODE REVIEW

File: src/core/Zones.h [hunk 1]
```diff
@@ -106,7 +106,7 @@ class CTheZones
 	static int16 FindAudioZone(CVector *pos);
 	static eLevelName FindZoneForPoint(const CVector &pos);
 	static CZone *GetPointerForZoneIndex(int32 i) { return i == -1 ? nil : &ZoneArray[i]; }
-	static int32 GetIndexForZonePointer(CZone *zone) { return zone == nil ? -1 : zone - ZoneArray; }
+	static uintptr GetIndexForZonePointer(CZone *zone) { return zone == nil ? -1 : zone - ZoneArray; }
```
Comment (blingu) [inline src/core/Zones.h:109]: I think an unsigned return type doesn't belong here, as it's possible to return -1

File: rwsdk/include/d3d8/rwplcore.h [hunk 2]
```diff
@@ -1567,7 +1567,7 @@ struct RwSurfaceProperties
 #define rwPIOVER2 (rwPI / (RwReal)(2.0f))
 #endif
 #define RWRGBALONG(r,g,b,a)                                             \
-    ((RwUInt32) (((a) << 24) | ((r) << 16) | ((g) << 8) | (b)))
```
Comment (aap) [inline rwsdk/include/d3d8/rwplcore.h:1570]: no

File: src/core/Zones.h (hunk 1 above)
Comment (aap) [reply src/core/Zones.h:109]: agreed. int32 is correct here. technically ssize_t would be the correct type but that's not portable and we don't have 2 billion zones anyway

File: src/core/templates.h [hunk 3]
```diff
@@ -127,9 +127,9 @@ class CPool
 		int i = GetJustIndex(entry);
 		return m_flags[i].u + (i<<8);
 	}
-	int GetJustIndex(T *entry){
+	uintptr GetJustIndex(T *entry){
 		// TODO: the cast is unsafe
-		return (int)((U*)entry - m_entries);
+		return (uintptr)((U*)entry - m_entries);
```
Comment (aap) [inline src/core/templates.h:132]: this is size_t

File: src/rw/VisibilityPlugins.cpp [hunk 4]
```diff
@@ -211,7 +211,7 @@ CVisibilityPlugins::RenderObjNormalAtomic(RpAtomic *atomic)
 }
 
 RpAtomic*
-CVisibilityPlugins::RenderAlphaAtomic(RpAtomic *atomic, int alpha)
+CVisibilityPlugins::RenderAlphaAtomic(RpAtomic *atomic, uintptr alpha)
```
Comment (aap) [inline src/rw/VisibilityPlugins.cpp:214]: what? why?

File: src/rw/VisibilityPlugins.cpp [hunk 5]
```diff
@@ -231,7 +231,7 @@ CVisibilityPlugins::RenderFadingAtomic(RpAtomic *atomic, float camdist)
 {
 	RpAtomic *lodatm;
 	float fadefactor;
-	uint32 alpha;
+	uintptr alpha;
```
Comment (aap) [inline src/rw/VisibilityPlugins.cpp:234]: again, why?

File: src/rw/VisibilityPlugins.cpp [hunk 6]
```diff
@@ -802,7 +802,7 @@ CVisibilityPlugins::SetFrameHierarchyId(RwFrame *frame, int32 id)
 	FRAMEEXT(frame)->id = id;
 }
 
-int32
+uintptr
 CVisibilityPlugins::GetFrameHierarchyId(RwFrame *frame)
```
Comment (aap) [inline src/rw/VisibilityPlugins.cpp:806]: no. this is an int32

File: src/vehicles/Automobile.cpp [hunk 7]
```diff
@@ -4542,7 +4542,7 @@ SetVehicleAtomicVisibilityCB(RwObject *object, void *data)
 }
 
 void
-CAutomobile::SetComponentVisibility(RwFrame *frame, uint32 flags)
+CAutomobile::SetComponentVisibility(RwFrame *frame, uintptr flags)
```
Comment (aap) [inline src/vehicles/Automobile.cpp:4545]: also wrong. it's an int

**Comment by aap** [comment]
this is all wrong

File: src/core/Zones.h (hunk 1 above)
Comment (erorcun) [reply src/core/Zones.h:109]: intptr_t is portable - or re3 has definitions of it

**Pull request closed without merging**